### PR TITLE
Test license selection in Settings

### PR DESCRIPTION
**Pull request by whym**
Tests like this might help finding problems like #506 earlier and preventing regression.

### CODE REVIEW

**Comment by whym** [comment]
Looks like I need to make some more Travis tweaks.

**Comment by whym** [comment]
I've managed to (kind of) fix it.

File: app/src/androidTest/java/fr/free/nrw/commons/SettingsActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        // click "License" (the first item)
+        Espresso.onData(anything())
+                .inAdapterView(findPreferenceList())
+                .atPosition(0)
+                .perform(ViewActions.click());
+
+        // click "CC BY-4.0"
+        Espresso.onView(
+                // FIXME: just R.string.license_name_cc_by_four should be fine but fails on Travis
+                textAnyOf(R.string.license_name_cc_by_four, R.string.license_name_cc_by_4_0)
```
Comment (whym) [inline app/src/androidTest/java/fr/free/nrw/commons/SettingsActivityTest.java]: Not entirely happy with this solution of accepting the two strings, but it should do no effective harm.

**Comment by misaochan** [comment]
@whym , sorry for the newbie question, but how do we use this?

**Comment by whym** [comment]
If you open SettingsActivityTest.java you will see triangles next to some line numbers. Click one of them to run the corresponding test(s).

Also, this patch includes changes to the Travis CI settings so that the tests are run automatically there.

**Comment by misaochan** [comment]
Awesome, thanks!

**Pull request merged**